test/index.js

```diff
@@ -17,32 +14,30 @@ var testPages = fs.readdirSync(testPageRoot).map(function(dir) {
 });
 
 describe("Test page", function() {
-  var oldLibPath = process.env.READABILITY_LIB_PATH;
-  process.env.READABILITY_LIB_PATH = path.join(__dirname, "..", "Readability.js");
-  var scrape = require("readable-proxy").scrape;
   testPages.forEach(function(testPage) {
     describe(testPage.dir, function() {
-      it("should render as expected", function() {
-        // Allows up to 10 seconds for parsing to complete.
-        // XXX: Scraping is damn slow. Investigate.
-        this.timeout(10000);
+      it("should render as expected", function(done) {
         var expected = fs.readFileSync(testPage.expected, {encoding: "utf-8"});
-        return scrape("file://" + testPage.source).catch(function(err) {
-          throw err;
-        }).then(function(result) {
-          // print Readability log messages
-          (result.consoleLogs || [])
-            .filter(function(logMessage) {
-              return logMessage.indexOf("Reader: (Readability)") === 0;
-            })
-            .forEach(function(logMessage) {
-              console.log("[LOG]", logMessage);
-            });
-          // normalize html
-          return prettyPrint(result.content);
-        }).should.eventually.become(prettyPrint(expected));
+        jsdom.env(
+          fs.readFileSync(testPage.source, {encoding: "utf-8"}),
+          [path.join(__dirname, "..", "Readability.js")],
+          function (errors, window) {
+            if (errors) {
+              throw new Error(errors);
+            }
+            var uri = {
+              spec: "http://fakehost/test/page.html",
+              host: "fake-host",
```

Comment:
![](http://readeroffictions.com/wp-content/uploads/2014/10/gif-robert-de-niro-watching-you.gif)